sys/vm/vm_page.c

```diff
@@ -1820,6 +1820,15 @@ vm_page_replace_hold(vm_page_t mnew, vm_object_t object, vm_pindex_t pindex,
 	 */
 	if (pmap_page_is_write_mapped(mnew))
 		vm_object_set_writeable_dirty(object);
+
+	/*
+	 * Copy across capability tracking flags
+	 */
+	if (vm_page_astate_load(mold).flags & PGA_CAPDIRTY)
+		vm_page_capdirty(mnew);
+	if (mold->oflags & VPO_CAPSTORE)
+		mnew->oflags |= VPO_CAPSTORE;
```

Comment:
Based on what's happening above for VPO_UNMANAGED, and the fact that nothing else is directly copying things from `mold` to `mnew`, should it not be up to the caller to have set the flags correctly?

Reply:
I can't say I know for sure, but that sounds reasonable.  Lemme drop this hunk and see what happens.